### PR TITLE
ui,build: write typescript declarations to multiple directories in watch mode

### PR DESCRIPTION
**Pull request by sjbarag**
Previously, running `pnpm tsc:watch` for incremental type-checking of `cluster-ui` would only write files to `./dist/types/*`. This worked fine, but the introduction of a push model for webpack output[^1] meant it was possible to have compiled JS without declarations in some other directory. Push typescript declarations into multiple directories as well, using a small integration with the TypeScript compiler API[^2].

[^1]: 7a6ec9565c0 (ui,build: push cluster-ui assets into external folder during watch mode, 2023-07-24)
[^2]: https://github.com/microsoft/TypeScript/wiki/Using-the-Compiler-API/167d197d290bec04b626b91b6f453123ef309e58#writing-an-incremental-program-watcher

Release note: None
Epic: none

### CODE REVIEW

**Comment by sjbarag** [comment]
Still a draft PR so I can get the license comment at the top of `build/util/index.js`

File: pkg/ui/workspaces/cluster-ui/src/index.ts [hunk 1]
```diff
@@ -56,3 +56,7 @@ export * from "./selectors";
 export * from "./contexts";
 export * from "./timestamp";
 export * from "./databases";
+
+export function lol() {
```
Comment (fantapop) [inline pkg/ui/workspaces/cluster-ui/src/index.ts:60]: Is this file intended to remain in the PR?

**Review by fantapop — APPROVED** [review]
Looks good to me. I saw a file that looked like it was maybe left in by mistake.  Thanks for doing this!

**Review by nathanstilwell — APPROVED** [review]

**Comment by sjbarag** [comment]
Thanks for the reviews, friends!

bors r=fantapop,nathanstilwell

**Comment by sjbarag** [comment]
Right, forgot to add `"build/util/**"` to the webpack glob in `pkg/ui/workspaces/cluster-ui/BUILD.bazel`. I'll do that after lunch :)
